src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/repository/VisitRepository.kt

```diff
@@ -276,4 +276,31 @@ interface VisitRepository : JpaRepository<Visit, Long>, JpaSpecificationExecutor
       " ORDER BY v.sessionSlot.slotStart,v.id",
   )
   fun findBookedVisitsForDate(prisonCode: String, date: LocalDate): List<Visit>
+
+  @Query(
+    "Select v.reference FROM visit v " +
+      " LEFT JOIN event_audit ea on ea.booking_reference = v.reference and ea.type = 'BOOKED_VISIT' " +
+      " LEFT JOIN actioned_by ab on ab.id = ea.actioned_by_id" +
+      " LEFT JOIN session_slot ss on ss.id = v.session_slot_id " +
+      " WHERE ab.booker_reference = :bookerReference AND v.visit_status = 'BOOKED' AND ss.slot_date >= CURRENT_DATE AND " +
+      "      v.user_type = 'PUBLIC' GROUP BY v.reference",
+    nativeQuery = true,
+  )
+  fun getPublicFutureBookingReferenceByBookerReference(bookerReference: String): List<String>
+
+  @Query(
+    "Select v.reference FROM visit v " +
+      " LEFT JOIN event_audit ea on ea.booking_reference = v.reference and ea.type = 'CANCELLED_VISIT' " +
```

Comment:
sorry just noticed this but do we need visits with ea.type = 'BOOKED_VISIT' so that visits booked by the booker are returned?

Reply:
yes good spot..